scripts/rebuild-endpoints.py

```diff
@@ -39,240 +30,89 @@
-    #   class provides iterators that allow you to iterate over all known
-    #   regions for all known partitions).
-    # * Helper method for static hostname lookup by abstracting the
-    #   sslCommonName checks into a "get_hostname" method.
-    # * Allowing you to use "service names" specific to boto2 when
-    #   generating endpoints.  Internally this has a mapping of which endpoint
-    #   prefixes to use.
-
-    SERVICE_RENAMES = {
-        # The botocore resolver is based on endpoint prefix.
-        # These don't always sync up to the name that boto2 uses.
-        # A mapping can be provided that handles the mapping between
-        # "service names" and endpoint prefixes.
-        'awslambda': 'lambda',
-        'cloudwatch': 'monitoring',
-        'ses': 'email',
-        'ec2containerservice': 'ecs',
-        'configservice': 'config',
-    }
-
-    def __init__(self, resolver, endpoint_data,
-                 service_name_map=None):
-        #: An instance of botocore.regions.EndpointResolver.
-        self._resolver = resolver
-        self._endpoint_data = endpoint_data
-        if service_name_map is None:
-            service_name_map = self.SERVICE_RENAMES
-        self._service_map = service_name_map
-
-    def regions_for_service(self, service_name):
-        # "What are all the regions EC2 is in across all known partitions?"
-        endpoint_prefix = self._endpoint_prefix(service_name)
-        for partition_name in self.get_available_partitions():
-            if self.is_service_in_partition(service_name, partition_name):
-                for region_name in self._resolver.get_available_endpoints(
-                        endpoint_prefix, partition_name):
-                    yield region_name
-
-    def regions_for_partition(self, partition_name):
-        # "What are all the known regions in a given partition?"
-        # This is used in boto to create entries for "cloudfront"
-        # for every region:
-        # us-east-1: cloudfront.amazonaws.com
-        # us-west-2: cloudfront.amazonaws.com
-        # ...
-        partition_data = self._get_partition_data(partition_name)
-        return [r for r in list(partition_data['regions'])
-                if 'global' not in r]
-
-    def partitions_for_service(self, service_name):
-        # "In which partitions is 'cloudfront' available?"
-        # This is used because we should *not* generate entries
-        # for cn-north-1 for cloudfront, it's not available in China.
-        # This can be accomplished by using this method and
-        # regions_for_partition. See the _special_case_global_service
-        # method in StaticEndpointBuilder.
-        for partition_name in self.get_available_partitions():
-            if self.is_service_in_partition(service_name, partition_name):
-                yield partition_name
-
-    def get_available_partitions(self):
-        return self._resolver.get_available_partitions()
+SUPPORTED_SERVICES = [
```

Comment:
Might be good to point out that this is botocore naming scheme not boto2 naming scheme. That tripped me up a bit.